### ISSUE TITLE
Support paid apps in non-curated repos

### ISSUE THREAD

**Issue by jhaygood86** [elementary/appcenter#1707]
### Problem

Currently, AppCenter only allows paid apps from curated repos. If the appstream metadata indicates that the app is a paid app, it should be supported regardless of the repo it's from

### Proposal

Support all paid app features regardless if using appcenter remote or another remote.

**Comment by danrabbit** [comment]
It's not really possible since we need a stripe key that is specific to our store in order to process transactions through stripe.

What we have talked about doing though is surfacing the "Donate" URL from appstream more boldly so that app developers can monetize that way even if the app transaction itself isn't supported

**Comment by jhaygood86** [comment]
In my case, the app **does** have the appcenter specific stripe key in the appstream metadata. AppCenter just ignores it because its not in the appcenter remote.

**Comment by jhaygood86** [comment]
Thinking about it, the ask is really to allow application authors to opt their non-curated applications into appcenter payments regardless if they are in the appcenter remote.

I'd be fine if that required a registration process to work to register the flatpak remote and application id so that elementary knows the application author is requesting it.

**Comment by jhaygood86** [comment]
Hm... I'm not specifically tied to elementary accepting the payments, just on making said payments as easy and obvious as they are for curated apps, i.e., you don't see "Free" in the download page in appcenter, if the app isn't purchased, it shows a price, etc...

**Comment by davidmhewitt** [comment]
Yeah, if you're hosting your own remote, you're absolutely free to host your own authenticator in there.

And once we've build the AppCenter authenticator, it'd be open source, so you could fork it and host your own differently branded version of it on your own remote and that'd all work through the AppCenter desktop application since it's not AppCenter that handles the payment dialog anymore. You click "Install" on an app from any remote in AppCenter, and if that remote has been configured to use an Authenticator, then it spawns the authenticator flow.

**Comment by davidmhewitt** [comment]
I'm going to close this because I don't think there's anything actionable that we can/want to do with AppCenter here.

There's already an issue open to surface AppStream donation links more prominently: https://github.com/elementary/appcenter/issues/1516

And once the planned work to build an authenticator for the elementary remote goes ahead, AppCenter will support authenticator flows too.
